database/settings/settings.go

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package settings
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/lib/pq"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+
+	"github.com/go-vela/server/api/types/settings"
+	"github.com/go-vela/server/util"
+)
+
+const (
+	TableSettings = "settings"
+)
+
+var (
+	// ErrEmptyCloneImage defines the error type when a
+	// Settings type has an empty CloneImage field provided.
+	ErrEmptyCloneImage = errors.New("empty settings clone image provided")
+)
+
+type (
+	// config represents the settings required to create the engine that implements the SettingsInterface interface.
+	config struct {
+		// specifies to skip creating tables and indexes for the Settings engine
+		SkipCreation bool
+	}
+
+	// engine represents the settings functionality that implements the SettingsInterface interface.
+	engine struct {
+		// engine configuration settings used in settings functions
+		config *config
+
+		ctx context.Context
+
+		// gorm.io/gorm database client used in settings functions
+		//
+		// https://pkg.go.dev/gorm.io/gorm#DB
+		client *gorm.DB
+
+		// sirupsen/logrus logger used in settings functions
+		//
+		// https://pkg.go.dev/github.com/sirupsen/logrus#Entry
+		logger *logrus.Entry
+	}
+
+	// Platform is the database representation of platform settings.
+	Platform struct {
+		ID sql.NullInt64 `sql:"id"`
+		Compiler
+		Queue
+
+		RepoAllowlist     pq.StringArray `json:"repo_allowlist" sql:"repo_allowlist" gorm:"type:varchar(1000)"`
+		ScheduleAllowlist pq.StringArray `json:"schedule_allowlist" sql:"schedule_allowlist" gorm:"type:varchar(1000)"`
+
+		CreatedAt sql.NullInt64  `sql:"created_at"`
+		UpdatedAt sql.NullInt64  `sql:"updated_at"`
+		UpdatedBy sql.NullString `sql:"updated_by"`
+	}
+
+	// Compiler is the database representation of compiler settings.
+	Compiler struct {
+		CloneImage        sql.NullString `json:"clone_image" sql:"clone_image"`
+		TemplateDepth     sql.NullInt64  `json:"template_depth" sql:"template_depth"`
+		StarlarkExecLimit sql.NullInt64  `json:"starlark_exec_limit" sql:"starlark_exec_limit"`
+	}
+
+	// Queue is the database representation of queue settings.
+	Queue struct {
+		Routes pq.StringArray `json:"routes" sql:"routes" gorm:"type:varchar(1000)"`
+	}
```

Comment:
Can these be moved to `database/types` for consistency?

Reply:
done